yt_dlp/extractor/triller.py

```diff
@@ -0,0 +1,339 @@
+        if not user_info:
+            self.report_warning('Unable to extract user info')
+
+        if user_info.get('private') and not user_info.get('followed_by_me'):
+            raise ExtractorError('This video is private', expected=True)
+
+        if user_info.get('blocked_by_user') or user_info.get('blocking_user'):
+            raise ExtractorError('The author of the video is blocked', expected=True)
+
+        return {
+            **self._parse_video_info(video_info, user_info),
+        }
+
+
+class TrillerUserIE(TrillerBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?triller\.co/@(?P<id>[\w\._]+)/?(?:$|[#?])'
+    _TESTS = [{
+        # first videos request only returns 2 videos
+        'url': 'https://triller.co/@theestallion',
+        'playlist_mincount': 9,
+        'info_dict': {
+            'id': '18992236',
+            'title': 'theestallion',
+            'thumbnail': r're:^https://uploads\.cdn\.triller\.co/.+\.jpg$',
+        }
+    }, {
+        'url': 'https://triller.co/@charlidamelio',
+        'playlist_mincount': 25,
+        'info_dict': {
+            'id': '1875551',
+            'title': 'charlidamelio',
+            'thumbnail': r're:^https://uploads\.cdn\.triller\.co/.+\.jpg$',
+        }
+    }]
+
+    def _extract_video_list(self, username, user_id, limit=6):
+        query = {
+            'limit': limit,
+        }
+        for page in itertools.count(1):
+            for retry in self.RetryManager():
+                try:
+                    video_list = self._download_json(
+                        f'{self._API_BASE_URL}/api/users/{user_id}/videos',
+                        username, note=f'Downloading user video list page {page}',
+                        errnote='Unable to download user video list', headers={
+                            'Authorization': f'Bearer {self._AUTH_TOKEN}',
+                            'Origin': 'https://triller.co',
+                        }, query=query)
+                except ExtractorError as e:
+                    if isinstance(e.cause, json.JSONDecodeError) and e.cause.pos == 0:
+                        retry.error = e
+                        continue
+                    raise
+            yield from video_list.get('videos', [])
+            if not video_list.get('videos'):
+                break
+            before_time = traverse_obj(video_list, ('videos', -1, 'timestamp'))
+            if not before_time:
+                break
+            query = {
+                'before_time': before_time,
+                'limit': limit,
+            }
+
+    def _entries(self, videos, user_info):
+        for video in videos:
+            yield {
+                **self._parse_video_info(video, user_info),
+                'extractor_key': TrillerIE.ie_key(),
+                'extractor': 'Triller',
+            }
+
+    def _real_extract(self, url):
+        username = self._match_id(url)
+        if not self._IS_LOGGED_IN:
```

Comment:
```suggestion
        if not self._AUTH_TOKEN:
```
_IS_LOGGED_IN is not even needed as far as I can tell

Reply:
Also, this should be in _real_initialize imo

Reply:
yeah `_IS_LOGGED_IN` is redundant, removed

moved `_create_guest` into the `_real_initialize` function for `TrillerUserIE` -- an auth token is only required when scraping user profiles, not individual videos